roles/controller_casc_from_aap/tasks/all.yml

```diff
@@ -1,7 +1,23 @@
 ---
+- name: "Get the Authentication Token for the future requests"
+  uri:
+    url: "https://{{ controller_hostname }}/api/v2/tokens/"
+    user: "{{ controller_username }}"
+    password: "{{ controller_password }}"
+    method: POST
+    force_basic_auth: true
+    validate_certs: "{{ controller_validate_certs }}"
+    status_code: 201
+  register: authtoken_res
+
+- name: "Set the oauth token to be used since now"
+  set_fact:
+    oauthtoken: "{{ authtoken_res.json.token }}"
+  no_log: true
+
```

Comment:
It can be a good idea to add an end task in this file to remove the oauth token generated.